src/pages/City/CityPage.tsx

```diff
@@ -19,27 +20,21 @@ const CityPage = () => {
   const normalizedCity = useMemo(() => convertSpecialCharacters(city), [city]);
 
   const { data, isLoading, isError } = Weather.useLocation(normalizedCity);
-  const { state }: NavigateOptions = useLocation();
 
   useEffect(() => {
     if (isError) {
       navigate(Paths.Map);
     }
   }, [isError, navigate]);
 
-  const renderCityName = () => {
-    if (isLoading) {
-      return <CityNameLoader />;
-    }
-    if (data) {
-      return <CityName>{state?.cityName || data.location.name}</CityName>;
-    }
-  };
+  const cityName = normalizedCity.split(',')[0].charAt(0).toUpperCase() + normalizedCity.split(',')[0].slice(1);
+  const coordinates = data && serializeCoordinates({ latitude: data?.location.lat, longitude: data?.location.lon });
 
   return (
     <PageContentContainer data-testid={CityPageTestIds.Container}>
-      {renderCityName()}
+      <CityNameWidget loading={isLoading} cityName={data !== undefined ? data.location.name : undefined} />
```

Comment:
**suggestion** I think it can be replaced with  `cityName={data?.location.name}`